src/test/pmem2_mem_ext/TESTS.py

```diff
@@ -0,0 +1,299 @@
+ALL_FLAGS = PMEM_F_MEM_NODRAIN | PMEM_F_MEM_NOFLUSH |
+            PMEM_F_MEM_NONTEMPORAL | PMEM_F_MEM_TEMPORAL |
+            PMEM_F_MEM_WC | PMEM_F_MEM_WB
+'''
+ALL_FLAGS = 7
+
+# This match is valid for all cases with a BYTE
+# granularity and small (<256) data size.
+BYTE_MATCH_SMALL = \
+    [
+        [NO_FLAGS,                                      [128, "t"]],
+        [PMEM_F_MEM_NONTEMPORAL,                        [128, "nt"]],
+        [PMEM_F_MEM_TEMPORAL,                           [128, "t"]],
+        [PMEM_F_MEM_NONTEMPORAL_v_PMEM_F_MEM_TEMPORAL,  [128, "nt"]],
+        [PMEM_F_MEM_WC,                                 [128, "t"]],
+        [PMEM_F_MEM_WB,                                 [128, "t"]],
+        [PMEM_F_MEM_NOFLUSH,                            [128, "t"]],
+        [ALL_FLAGS,                                     [128, "t"]]
+    ]
+
+# This match is valid for all cases with a BYTE
+# granularity and big (>256) data size.
+BYTE_MATCH_BIG = \
+    [
+        [NO_FLAGS,                                      [1024, "t"]],
+        [PMEM_F_MEM_NONTEMPORAL,                        [1024, "nt"]],
+        [PMEM_F_MEM_TEMPORAL,                           [1024, "t"]],
+        [PMEM_F_MEM_NONTEMPORAL_v_PMEM_F_MEM_TEMPORAL,  [1024, "nt"]],
+        [PMEM_F_MEM_WC,                                 [1024, "t"]],
+        [PMEM_F_MEM_WB,                                 [1024, "t"]],
+        [PMEM_F_MEM_NOFLUSH,                            [1024, "t"]],
+        [ALL_FLAGS,                                     [1024, "t"]]
+    ]
+
+# This match is valid for all cases with a PAGE/CACHELINE
+# granularity and small (<256) data size.
+PAGE_CACHELINE_SMALL = \
+    [
+        [NO_FLAGS,                                      [128, "t"]],
+        [PMEM_F_MEM_NONTEMPORAL,                        [128, "nt"]],
+        [PMEM_F_MEM_TEMPORAL,                           [128, "t"]],
+        [PMEM_F_MEM_NONTEMPORAL_v_PMEM_F_MEM_TEMPORAL,  [128, "nt"]],
+        [PMEM_F_MEM_WC,                                 [128, "nt"]],
+        [PMEM_F_MEM_WB,                                 [128, "t"]],
+        [PMEM_F_MEM_NOFLUSH,                            [128, "t"]],
+        [ALL_FLAGS,                                     [128, "t"]]
+    ]
+
+# This match is valid for all cases with a PAGE/CACHELINE
+# granularity and big (>256) data size.
+PAGE_CACHELINE_BIG = \
+    [
+        [NO_FLAGS,                                      [1024, "nt"]],
+        [PMEM_F_MEM_NONTEMPORAL,                        [1024, "nt"]],
+        [PMEM_F_MEM_TEMPORAL,                           [1024, "t"]],
+        [PMEM_F_MEM_NONTEMPORAL_v_PMEM_F_MEM_TEMPORAL,  [1024, "nt"]],
+        [PMEM_F_MEM_WC,                                 [1024, "nt"]],
+        [PMEM_F_MEM_WB,                                 [1024, "t"]],
+        [PMEM_F_MEM_NOFLUSH,                            [1024, "t"]],
+        [ALL_FLAGS,                                     [1024, "t"]]
+    ]
+
+test_cases = \
```

Comment:
do we really need a separate table for that? can't we just store in TEST$N classes?